pwa/src/components/endpoints/endpointForm.tsx

```diff
@@ -5,140 +5,116 @@ import {
-      })
-      .catch((error) => {
-        setShowSpinner(false);
-        console.error("Error:", error);
-        setAlert(null);
-        setAlert({type: 'danger', message: error.message});
-      });
 
+  const handleSetEndpoints = () => {
+    setShowSpinner(true)
+
+    API.Endpoint.getOne(id)
+      .then((res) => {
+        setEndpoint(res.data)
+      })
+      .catch((err) => {
+        throw new Error('GET Endpoint error: ' + err)
+      })
+      .finally(() => {
+        setShowSpinner(false)
+      })
   }
 
-  const getApplications = () => {
-    setShowSpinner(true);
-    fetch(`${context.adminUrl}/applications`, {
-      credentials: "include",
-      headers: {"Content-Type": "application/json", 'Authorization': 'Bearer ' + sessionStorage.getItem('jwt')},
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        setShowSpinner(false);
-        if (data['hydra:member'] !== undefined && data['hydra:member'].length > 0) {
-          setApplications(data['hydra:member']);
-        }
+  const handleSetApplications = () => {
+    setShowSpinner(true)
+
+    API.Application.getAll()
+      .then((res) => {
+        setApplication(res.data)
       })
-      .catch((error) => {
-        setShowSpinner(false);
-        console.error("Error:", error);
-        setAlert(null);
-        setAlert({type: 'danger', message: error.message});
-      });
-  };
+      .catch((err) => {
+        throw new Error('GET application error: ' + err)
+      })
+      .finally(() => {
+        setShowSpinner(false)
+      })
+  }
 
   const saveEndpoint = (event) => {
     event.preventDefault();
     setLoadingOverlay(true);
 
     let body: {} = {
       name: event.target.name.value,
-      description: event.target.description.value ? event.target.description.value : null,
+      description: event.target.description.value
+        ? event.target.description.value : null,
       type: event.target.type.value,
       path: event.target.path.value,
-      application: event.target.application.value ? event.target.application.value : null,
+      application: event.target.application.value
+        ? event.target.application.value : null,
     };
 
     body = removeEmptyObjectValues(body);
+
     if (!checkValues([body["name"], body["type"], body["path"]])) {
-      setAlert(null);
-      setAlert({type: 'danger', message: 'Required fields are empty'});
-      setShowSpinner(false);
       return;
     }
 
-    let url = `${context.adminUrl}/endpoints`;
-    let method = "POST";
-    if (id !== "new") {
-      url = `${url}/${id}`;
-      method = "PUT";
+
+    if (!id) { // unset id means we're creating a new entry
+      API.Endpoint.create(body)
+        .then((res) => {
+          setEndpoint(res.data)
+          navigate('/endpoints/${id}')
+        })
+        .catch((err) => {
+          setAlert({type: 'danger', message: err.message});
+          throw new Error('Create endpoint error: ' + err)
+        })
```

Comment:
After the `.catch()` we still need a `finally()` to stop the `loadingOverlay`, this can be implemented as followed:

```
.finally(() => { setLoadingOverlay(false) })
```